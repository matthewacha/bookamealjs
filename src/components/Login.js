import React from 'react';
import './static/staticHome.css';
import { Checkbox, FormGroup} from 'react-bootstrap';
import { PropTypes} from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { logIn, loginAdmin, signAdmin } from '../actions/credActions';
import { GetMeals } from '../actions/adminActions';

class Login extends React.Component{
	state = {checked:false}

	componentWillReceiveProps(recievedMessage){
		if(recievedMessage){
			if(recievedMessage.adminlog){
			// if login as caterer
				if (recievedMessage.adminlog.message){
					console.log(recievedMessage.adminlog.message);
					// flash not admin message
				}else if(recievedMessage.adminlog.token){
					localStorage.setItem('K_access_token', recievedMessage.adminlog.token);
					this.props.history.push("/adminDash");
				}
			}else if(recievedMessage.userlog){
				// if login as user
				console.log(recievedMessage)
					if (recievedMessage.userlog.message){
						console.log(recievedMessage.userlog.message);
						// flash unauthorized login
					}else if(recievedMessage.userlog.token){
						localStorage.setItem('access_token', recievedMessage.userlog.token);
						this.props.history.push("/userDash");
					}
				}
		}
	};

	circularStringify = (object) =>{
		let simpleObj={};
		for (let prop in object){
				if (!object.hasOwnProperty(prop)){
						continue;
				}
				if (typeof(object[prop]) === 'object'){
						continue;
				}
				simpleObj[prop] = object[prop];
		}
		return JSON.stringify(simpleObj)

	}

	onSubmit = e =>{
		e.preventDefault();
		let credentials = {
			email: e.target.elements.email.value,
			password:e.target.elements.password.value,
			isadmin:e.target.elements.isadmin.checked};
		if(String(credentials.isadmin)==='true'){
			console.log('administrator')
			this.props.logIn(JSON.stringify(credentials))
			console.log(localStorage.getItem('access_token'))
			this.props.loginAdmin(JSON.stringify(credentials));
			console.log(localStorage.getItem('K_access_token'))
		}else{
			console.log('not administrator')
			this.props.logIn(JSON.stringify(credentials))
		};
	}
  
	render(){
		return(
    <div>
        <nav>
		<div className="navibar">
			<ul>
				<li><a className="nav-log" href="Signup">Book-A-Meal</a></li>
				<li id="right-but"><a className="nav-logo" href="Signup">Signup</a></li>
			</ul>
	     </div>
	</nav>
	
	<div className = "background">
		<div className="input-forms">
			<div className="h2">
			<h2 id = "title">Login</h2>
			</div>
			<form onSubmit = { this.onSubmit }>
				<div className="forms-inputs">
					<div>
					  <label className="label">Email</label><br/>
					  <input type="email" name = "email" className="form-controls" placeholder="you@email.com" required/><br/><br/>
				      <label className="label">Password</label><br/>
					  <input type="password" name = "password" className="form-controls" placeholder="*********" required/><br/>
					  <Checkbox style ={{"color":"white"}} name = "isadmin" inline>Caterer</Checkbox>
					  </div><br/>
					<div>
					  <button className='submit-button' type="submit">Submit</button>
					</div>
				</div>
			</form>
		</div>
	</div>
    </div>
)}
}

Login.propTypes = {
	loginAdmin:PropTypes.func.isRequired,
	logIn: PropTypes.func.isRequired,
	userlog: PropTypes.object.isRequired,
	adminlog: PropTypes.object.isRequired,
	GetMeals:PropTypes.func.isRequired
}

const mapStateToProps = state => ({
	userlog: state.user.loggedData,
	adminlog: state.user.adminData,
	GetMeals:PropTypes.func.isRequired
});

export default withRouter(connect(mapStateToProps, { logIn, loginAdmin, GetMeals} )(Login));