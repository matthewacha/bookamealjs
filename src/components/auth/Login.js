import React from "react";
import "../static/staticHome.scss";
import { Checkbox } from "react-bootstrap";
import { notify } from "react-notify-toast";
import { PropTypes } from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import { logIn, loginAdmin } from "../../actions/credActions";
import { GetMeals } from "../../actions/adminActions";

class Login extends React.Component {
  componentWillReceiveProps(recievedMessage) {
    if (recievedMessage) {
      // if login as caterer
      if (recievedMessage.adminlog.message) {
        notify.show(recievedMessage.adminlog.message, "error");
      } else if (recievedMessage.adminlog.token) {
        notify.show("Successfully logged in", "success");
        this.props.history.push("/adminDash");
      } else {
        // if login as user
        if (recievedMessage.userlog.message) {
          notify.show(recievedMessage.userlog.message, "error");
        } else if (recievedMessage.userlog.token) {
          notify.show("Successfully logged in", "success");
          localStorage.setItem("access_token", recievedMessage.userlog.token);
          this.props.history.push("/userDash");
        }
      }
    }
  }

  onSubmit = e => {
    e.preventDefault();
    let credentials = {
      email: e.target.elements.email.value,
      password: e.target.elements.password.value,
      isadmin: e.target.elements.isadmin.checked
	};
    if (String(credentials.isadmin) === "true") {
      this.props.loginAdmin(JSON.stringify(credentials));
    } else {
      this.props.logIn(JSON.stringify(credentials));
    }
  };

  render() {
    return (
      <div>
        <nav>
          <div className="navibar">
            <ul>
              <li>
                <a
                  className="nav-log"
                  href="Signup"
                  style={{ paddingTop: "0px" }}
                >
                  Book-A-Meal
                </a>
              </li>
              <li id="right-but">
                <a className="nav-logo" href="Signup">
                  Signup
                </a>
              </li>
            </ul>
          </div>
        </nav>

        <div className="background">
          <div className="input-forms">
            <div className="h2">
              <h2 id="title">Login</h2>
            </div>
            <form className="login" onSubmit={this.onSubmit}>
              <div className="forms-inputs">
                <div>
                  <label className="label" id="email-label">
                    Email
                  </label>
                  <br />
                  <input
                    type="email"
                    id="email"
                    name="email"
                    className="form-controls"
                    placeholder="you@email.com"
                    required
                  />
                  <br />
                  <br />
                  <label className="label" id="password-label">
                    Password
                  </label>
                  <br />
                  <input
                    type="password"
                    id="password"
                    name="password"
                    className="form-controls"
                    placeholder="*********"
                    required
                  />
                  <br />
                  <Checkbox
                    style={{ color: "white" }}
                    id="isadmin"
                    name="isadmin"
                    inline
                  >
                    Caterer
                  </Checkbox>
                </div>
                <br />
                <div>
                  <button className="submit-button" type="submit">
                    Submit
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

Login.propTypes = {
  loginAdmin: PropTypes.func.isRequired,
  logIn: PropTypes.func.isRequired,
  userlog: PropTypes.object.isRequired,
  adminlog: PropTypes.object.isRequired,
  GetMeals: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  userlog: state.user.loggedData,
  adminlog: state.user.adminData,
  GetMeals: PropTypes.func.isRequired
});

export default connect(
  mapStateToProps,
  { logIn, loginAdmin, GetMeals }
)(Login);
