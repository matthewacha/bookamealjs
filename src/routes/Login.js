import React from 'react';
import './static/staticHome.css';

export default () => (
    <div>
        <nav>
		<div className="navibar">
			<ul>
				<li><a className="nav-logo" href="Signup">Book-A-Meal</a></li>
				<li id="right-but"><a className="nav-logo" href="Signup">Signup</a></li>
			</ul>
	     </div>
	</nav>
	
	<div className = "background">
		<div className="input-forms">
			<div className="h2">
			<h2>Login</h2>
			</div>
			<form action="admin-dash.html" href="customerHome.html">
				<div className="form-inputs">
					<div>
					  <label className="label">Email</label><br/>
					  <input type="email" className="form-control" placeholder="you@email.com" required/><br/><br/>
				      <label className="label">Password</label><br/>
					  <input type="password" className="form-control" placeholder="*********" required/><br/>
					</div><br/>
					<div>
					  <button className='submit-button' type="submit">Submit</button>
					</div>
				</div>
			</form>
		</div>
	</div>
    </div>
);