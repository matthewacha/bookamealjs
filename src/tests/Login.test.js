import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import bookMeal from '../store';
import Login from '../components/Login';

describe('<Login/>', () => {
    it('includes a title Login', () => {
        let  signup = mount(<MemoryRouter initialEntries={['/login']} initialIndex={0}>
                                  <Login store = {bookMeal}/>
                              </MemoryRouter>);
        expect(signup.find('div.h2').text()).toEqual('Login');

        });

    it('there are two input fields', () => {
        let  signup = mount(<MemoryRouter initialEntries={['/login']} initialIndex={0}>
                                  <Login store = {bookMeal}/>
                              </MemoryRouter>);
        expect(signup.find('input.form-control').length).toEqual(2);

        });

    it('there are two labels', () => {
        let  signup = mount(<MemoryRouter initialEntries={['/login']} initialIndex={0}>
                                  <Login store = {bookMeal}/>
                              </MemoryRouter>);
        expect(signup.find('label.label').length).toEqual(2);

        });

    it('check that submit button redirects', () => {
        let  signup = mount(<MemoryRouter initialEntries={['/login']} initialIndex={0}>
                                  <Login store = {bookMeal}/>
                              </MemoryRouter>);
        expect(signup.find('button.submit-button').simulate('click',{target:{elements:{email:{value:"me@g.com"},password:{value:"animal"}}}}));
        //expect(signup).toMatchSnapshot();
        });

        it('check that navbar login button redirects', () => {
            let  signup = mount(<MemoryRouter initialEntries={['/login']} initialIndex={0}>
                                  <Login store = {bookMeal}/>
                              </MemoryRouter>);
            expect(signup.find('a.nav-logo').simulate('click'));
            //expect(signup).toMatchSnapshot();
            });
    }
)