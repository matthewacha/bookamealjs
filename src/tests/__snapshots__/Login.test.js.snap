// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> test form submits 1`] = `[Function]`;
